spec/default_spec.rb

```diff
@@ -9,6 +9,33 @@
       it 'converges successfully' do
         expect { chef_run }.to_not raise_error
       end
+      it do
+        expect(chef_run.node['varnish']['version']).to eq(3)
+      end
+      it do
+        expect(chef_run.node['varnish']['log_daemon']).to eq(false)
+      end
+      it do
+        expect(chef_run).to include_recipe('varnish')
+      end
+      it do
+        expect(chef_run).to create_template('/etc/sysconfig/varnishncsa').with(
+          source: 'varnishncsa.erb',
+          variables: { directives: chef_run.node['osl-varnish']['ncsa_directives'] }
```

Comment:
I think it would be better for you to statically set a set of test variables in this attribute in chefspec and then test it here and in the render_file part below too.